src/main/java/graphql/execution/directives/EncounteredDirectivesImpl.java

```diff
@@ -0,0 +1,89 @@
+package graphql.execution.directives;
+
+import graphql.Internal;
+import graphql.schema.GraphQLDirective;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import static graphql.Assert.assertNotNull;
+import static java.util.Collections.emptyList;
+import static java.util.stream.Collectors.toList;
+
+/**
+ * These objects are ALWAYS in the context of a single MergedField
+ */
+@Internal
+public class EncounteredDirectivesImpl implements EncounteredDirectives {
```

Comment:
Since it comes directly of the DataFetcherEnv then I think it makes sense to only give out an interface.

That way we can hide all the builders and so on and not make them API